src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -35,8 +35,13 @@ public async Task Invoke(HttpContext context)
 
             var endpoint = context.GetEndpoint();
 
-            // Flag to indicate to other systems, e.g. MVC, that authorization middleware was run for this request
-            context.Items[AuthorizationMiddlewareInvokedKey] = AuthorizationMiddlewareInvokedValue;
+            if (endpoint != null)
+            {
+                // Flag to indicate to the system that the middleware was run in the context of endpoint routing.
+                // Setting this flag allows a check in EndpointRoutingMiddleware that verifies if the middleware
+                // pipeline is wired correctly to succeed.
+                context.Items[AuthorizationMiddlewareInvokedKey] = AuthorizationMiddlewareInvokedValue;
```

Comment:
Rename key/value to indicate this is endpoint specific.

Reply:
Do you have any suggestions?

Reply:
EndpointAuthInvoked?

Reply:
AuthorizationInvokedWithEndpointsKey?